cli/cmd/describe.go

```diff
@@ -43,55 +46,7 @@ func wrapTextSuccessOfFailure(text string, success bool) string {
 	}
 }
 
-func cmdKindToK8sGVR(kind string) (schema.GroupVersionResource, error) {
-	kind = strings.ToLower(kind)
-	if kind == "deployment" || kind == "deployments" || kind == "dep" {
-		return schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"}, nil
-	}
-	if kind == "statefulset" || kind == "statefulsets" || kind == "sts" {
-		return schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "statefulsets"}, nil
-	}
-	if kind == "daemonset" || kind == "daemonsets" || kind == "ds" {
-		return schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "daemonsets"}, nil
-	}
-
-	return schema.GroupVersionResource{}, fmt.Errorf("unsupported kind: %s", kind)
-}
-
-func extractPodInfo(obj *unstructured.Unstructured) (*v1.PodTemplateSpec, string, error) {
-	gvk := obj.GroupVersionKind()
-
-	switch gvk.Kind {
-	case "Deployment":
-		var deployment appsv1.Deployment
-		err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &deployment)
-		if err != nil {
-			return nil, "", fmt.Errorf("failed to cast to Deployment: %v", err)
-		}
-		return &deployment.Spec.Template, metav1.FormatLabelSelector(deployment.Spec.Selector), nil
-
-	case "StatefulSet":
-		var statefulSet appsv1.StatefulSet
-		err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &statefulSet)
-		if err != nil {
-			return nil, "", fmt.Errorf("failed to cast to StatefulSet: %v", err)
-		}
-		return &statefulSet.Spec.Template, metav1.FormatLabelSelector(statefulSet.Spec.Selector), nil
-
-	case "DaemonSet":
-		var daemonSet appsv1.DaemonSet
-		err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &daemonSet)
-		if err != nil {
-			return nil, "", fmt.Errorf("failed to cast to DaemonSet: %v", err)
-		}
-		return &daemonSet.Spec.Template, metav1.FormatLabelSelector(daemonSet.Spec.Selector), nil
-
-	default:
-		return nil, "", fmt.Errorf("unsupported kind: %s", gvk.Kind)
-	}
-}
-
-func getInstrumentationLabelTexts(workload *unstructured.Unstructured, ns *v1.Namespace) (workloadText, nsText, decisionText string, instrumented bool) {
+func getInstrumentationLabelTexts(workload *K8sSourceObject, ns *v1.Namespace) (workloadText, nsText, decisionText string, instrumented bool) {
```

Comment:
Looks like we don't return anything in this function - so we can remove the return values from the function signature

Reply:
return values are used by name